### PR TITLE
Upgrade vitessce 0.1.10

### PR DESCRIPTION
**Pull request by ilan-gold**
If you want to test locally (any, some, or all - up to you):

MALDI: http://localhost:5001/browse/dataset/d66b802cbc2e8e83e84803f75d31bfcd
CODEX: http://localhost:5001/browse/dataset/63a35af025a3662cb368eb3999fe2948
MxIF: http://localhost:5001/browse/dataset/ac5d138c88d83b3cb80a4c42d36b05e2
PAS: http://localhost:5001/browse/dataset/2ce183e38a31ff1d836675d3b75654b0
SCRNA-seq: http://localhost:5001/browse/dataset/18c14ef9165c978a14256674158cc181

The IMS stuff is because these images have no data types except "image pyramid" so this (for now) is my best attempt at giving some sort of good configuration for them.

### CODE REVIEW

**Comment by mccalluc** [comment]
Haven't looked at the code yet, but some initial comments... most of which are not about Vitessce, but you might be able to forward them to the right people, if you think it's worth-while...
- CODEX and MALDI, the top row of pixels looks like it's been double sampled?
- Do you know why there's a checkerboard effect with the MALDI data?
- Would it be possible not to show the opacity- or time- or z-slider if there's only one option?
- When a layer is collapsed, it looks like the text of the title is centered... better to left align?
Screen Shot 2020-07-24 at 9 51 34 AM
- With the heatmap's hover indicator, I'm not sure whether it's the cell on the left or the right of the single pixel line.
- Is there any chance your polygon reduction code dropped at single vertex at the corner of the plot, and that that lower left triangle really should be inside the cell boundary?
Screen Shot 2020-07-24 at 9 59 19 AM
Screen Shot 2020-07-24 at 10 03 50 AM
- Where does the initial framing come from? http://localhost:5001/browse/dataset/2ce183e38a31ff1d836675d3b75654b0 in particular, zoomed in on just one corner.

Now to do a code review...

**Comment by ilan-gold** [comment]
>     * CODEX and MALDI, the top row of pixels looks like it's been double sampled?
Could you clarify this?
>     * Do you know why there's a checkerboard effect with the MALDI data?
This is an open bug, basically.  I'm reporting it to the Glencoe folks now.
>     * Would it be possible not to show the opacity- or time- or z-slider if there's only one option?
You're the second person to say this, so I think it's time to do that, yes.  Thanks.
>     * When a layer is collapsed, it looks like the text of the title is centered... better to left align?
This is a known bug with our code, but there is also basically no way to handle this in Material UI unfortunately.  I'm going to implement something custom, I think.
>     * With the heatmap's hover indicator, I'm not sure whether it's the cell on the left or the right of the single pixel line.
  I'll try to look closer.  Mark is going to update this layer so I may not look too closely.
>     * Is there any chance your polygon reduction code dropped at single vertex at the corner of the plot, and that that lower left triangle really should be inside the cell boundary?
I suppose that is possible, but I have seen other corners where this didn't happen so I think it might just be segmentation.  There doesn't look to be great DAPI data underneath, for example:
Screen Shot 2020-07-24 at 10 19 36 AM
>     * Where does the initial framing come from? http://localhost:5001/browse/dataset/2ce183e38a31ff1d836675d3b75654b0 in particular, zoomed in on just one corner.
I had not make an educated guess here based on types.  Going forward it would be best to do this dynamically based on the actual tiff (using rio-tiler, a python package).

**Comment by ilan-gold** [comment]
@mccalluc I was able to reproduce that last issue you mentioned, about zoom configuration, by first going to a MALDI url, and then going to a PAS url.  It's very odd, though, because when I log the `entity` out, it looks like PAS is still referring to some of the MALDI.  Looking into this now.

**Comment by ilan-gold** [comment]
@mccalluc I figured out what was going on.  It was an object mutability issue.  Pushing the fix now.

**Comment by mccalluc** [comment]
>>    * CODEX and MALDI, the top row of pixels looks like it's been double sampled?
>
> Could you clarify this?

Screen Shot 2020-07-24 at 11 25 13 AM

Maybe there's an off-by-one error upstream that's double counting the values in the first row? Or something like that?

> I suppose that is possible, but I have seen other corners where this didn't happen so I think it might just be segmentation. 

Not a big deal, but it's a literal "edge-case" that it might be worth telling the upstream developers about, It's curious that you say that sometimes it does work.

**Comment by ilan-gold** [comment]
> Not a big deal, but it's a literal "edge-case" that it might be worth telling the upstream developers about, It's curious that you say that sometimes it does work.
I meant in the sense that there are other corners that look totally normal: 
Screen Shot 2020-07-24 at 11 31 14 AM

File: context/app/api/vitessce.py [hunk 1]
```diff
@@ -233,7 +233,9 @@ def _build_vitessce_conf(self):
             return confs
```
[the remark this answers is not in the document]
Comment (ilan-gold) [reply context/app/api/vitessce.py:233]: I agree about the refactor.  I can change `confs` to `conf` as long as nothing is too broken by that.

File: context/app/api/vitessce.py [hunk 2]
```diff
@@ -233,7 +233,9 @@ def _build_vitessce_conf(self):
...
         elif self.assay_type in IMAGE_ASSAYS:
             found_images = _get_matches(file_paths_found, files[0]["rel_path"])
-            layer = self._build_multi_file_image_layer_conf(found_images)
+            # Do not show IMS images that are in the "/separate/" folder.
+            no_ims_separate_images = _get_matches(found_images, r"^((?!/separate/).)*$")
```
[the remark this answers is not in the document]
Comment (ilan-gold) [reply context/app/api/vitessce.py:237]: Good point.

**Comment by ilan-gold** [comment]
I can make an issue to refactor this.  I really didn't expect these many adhoc things, especially with the imaging modalities.

**Pull request merged**